src/layer/vox/controls.ts

```diff
@@ -0,0 +1,159 @@
+import { observeWatchable } from "#src/trackable_value.js";
+import { unpackRGB } from "#src/util/color.js";
+import { RefCounted } from "#src/util/disposable.js";
+import { vec3 } from "#src/util/geom.js";
+import { NullarySignal } from "#src/util/signal.js";
+import type { LayerControlDefinition } from "#src/widget/layer_control.js";
+import { registerLayerControl } from "#src/widget/layer_control.js";
+import { buttonLayerControl } from "#src/widget/layer_control_button.js";
+import { checkboxLayerControl } from "#src/widget/layer_control_checkbox.js";
+import { colorLayerControl } from "#src/widget/layer_control_color.js";
+import { enumLayerControl } from "#src/widget/layer_control_enum.js";
+import { rangeLayerControl } from "#src/widget/layer_control_range.js";
+
+class BigIntAsTrackableRGB
+  extends RefCounted
+  implements WatchableValueInterface<vec3>
+{
+  changed = new NullarySignal();
+  private tempColor = vec3.create();
+
+  constructor(public source: WatchableValueInterface<bigint>) {
+    super();
+    this.registerDisposer(source.changed.add(this.changed.dispatch));
+  }
+
+  get value(): vec3 {
+    const bigintValue = this.source.value;
+    const [r, g, b] = unpackRGB(Number(bigintValue & 0xffffffn));
+    vec3.set(this.tempColor, r, g, b);
+    return this.tempColor;
+  }
+
+  set value(newValue: vec3) {
+    const rgb = newValue.map((c: number) => Math.round(c * 255));
+    this.source.value = BigInt((rgb[0] << 16) | (rgb[1] << 8) | rgb[2]);
+  }
+}
+
+export const VOXEL_LAYER_CONTROLS: LayerControlDefinition<UserLayerWithVoxelEditing>[] =
+  [
+    {
+      label: "Brush size",
+      toolJson: { type: "vox-brush-size" },
+      ...rangeLayerControl((layer) => ({
+        value: layer.voxBrushRadius,
+        options: { min: 1, max: 64, step: 1 },
+      })),
+    },
+    {
+      label: "Eraser",
+      toolJson: { type: "vox-erase-mode" },
+      ...checkboxLayerControl((layer) => layer.voxEraseMode),
+    },
+    {
+      label: "Brush shape",
+      toolJson: { type: "vox-brush-shape" },
+      ...enumLayerControl(
+        (layer: UserLayerWithVoxelEditing) => layer.voxBrushShape,
+      ),
+    },
+    {
+      label: "Max fill voxels",
+      toolJson: { type: "vox-flood-max-voxels" },
+      ...rangeLayerControl((layer) => ({
+        value: layer.voxFloodMaxVoxels,
+        options: { min: 1, max: 1000000, step: 1000 },
+      })),
+    },
+    {
+      label: "Undo",
+      toolJson: { type: "vox-undo" },
+      ...buttonLayerControl({
+        text: "Undo",
+        onClick: (layer) =>
+          layer.handleVoxAction("undo", new LayerActionContext()),
+      }),
+    },
+    {
+      label: "Redo",
+      toolJson: { type: "vox-redo" },
+      ...buttonLayerControl({
+        text: "Redo",
+        onClick: (layer) =>
+          layer.handleVoxAction("redo", new LayerActionContext()),
+      }),
+    },
+    {
+      label: "Paint Color",
```

Comment:
If this is intended to show the color for segmentation layers, it shouldn't be editable, and also it doesn't always seem to stay in sync.